### PR TITLE
Initial PR

### LINKED ISSUE #73: [RSA Accumulator] Use Fermat Primality Test for verifying prime number
Instead of testing with listed numbers like [this solidity prime tester](https://github.com/matterinc/RSAAccumulator/blob/48b172a97c537014793582e85e78d7e399be2775/contracts/PrimeTester.sol), I propose other primaliry test [here](https://github.com/hskang9/verified-vyper-contracts/blob/master/contracts/accumulator/RSAAccumulator.vy).

in Fermat's primality test, if p is prime and a is an random integer,

```a**(p-1) mod p = 1```

For example, if a = 2, p = 17,

2**(17 - 1) mod 17 = 1 (prime)

Whereas a = 2, p = 16,
2**(16 - 1) mod 16  = 0 (not prime)

All I am worried for this algorithm is when calculating exponents which are bigger than ```uint256``` can handle, but I see there is already functions for handling that.

Is there any way that we can implement this algorithm? If there is a blocking, what is it?

### PR DESCRIPTION
**Pull request by hskang9**
## What I did

resolve #73 

I will extend this to have:

- solve easy answers(e.g. 2,3,5,7)
- Resolve Fermat liars if they exist

## Cute Animal Picture

pic

### CODE REVIEW

File: contracts/accumulator/isPrime.vy [hunk 1]
```diff
@@ -0,0 +1,58 @@
+# @dev isPrime
+# @author Nick Hyungsuk Kang (@hskang9)
+# Prime Tester using Fermat's primality test where a=2
+
+@private
+@constant
+def _compare(_a: uint256[8], _b: uint256[8]) -> int128:
+    for i in range(M_LIST_LENGTH):
```
Comment (nrryuya) [inline contracts/accumulator/isPrime.vy]: I got this error here ;)
```
line 8: Range only accepts literal (constant) values
    for i in range(M_LIST_LENGTH):
-------------------^
```

**Review by nrryuya — CHANGES_REQUESTED** [review]

**Comment by nrryuya** [comment]
@hskang9 
Thank you! A few comments.
- Could you rebase to the latest master branch and create a new PR? This inlined version is WIP and I think it'd be better to have a new directory and file `contracts/prime/isPrime.vy` because this contract can be used as stand-alone.
- Tests are always necessary ;) Please add tests at `tests/prime/test_isPrime.py`.

**Comment by nrryuya** [comment]
@hskang9 
About tests, I'll prepare `BigInt.vy` which can be used by `isPrime.vy`.

**Comment by hskang9** [comment]
Thank you! @nrryuya now that I have Big integer operations in separate file, I can focus on the tester. I will create WIP to master branch based on the comments received 👍

**Comment by hskang9** [comment]
Closing this for new PR

**Comment by nrryuya** [comment]
@hskang9 
#79 is still WIP so it's OK to wait for my work or to prepare only test cases (without actually executing them) ;)

**Comment by hskang9** [comment]
I think I will put TODO in comment to remove big integer operations after ```BigInt.vy``` is developed.

**Comment by hskang9** [comment]
This PR is now moved to #81

**Pull request closed without merging**